source/slang/emit.cpp

```diff
@@ -972,9 +972,36 @@ struct EmitVisitor
         case CodeGenTarget::GLSL_Vulkan:
         case CodeGenTarget::GLSL_Vulkan_OneDesc:
             {
-                emitGLSLTypePrefix(vecType->getElementType());
-                Emit("vec");
-                EmitVal(vecType->getElementCount());
+                // Need to special case if there is a single element in the vector
+                // as there is no such thing in glsl as vec1 
+                IRInst* elementCountInst = vecType->getElementCount();
+
+                if (elementCountInst->op != kIROp_IntLit)
+                {
+                    SLANG_DIAGNOSE_UNEXPECTED(getSink(), SourceLoc(), "Expecting an integral size for vector size");
```

Comment:
We should try to use `vecType` instead of `SourceLoc()` here, so that the user will get the source location associated with the type instead of no source location. It might be better than nothing.

If `vecType` alone doesn't work, you could try `vecType->loc`, although we should fix up the diagnostics-reporting code to support any `IRInst*` as a placeholder for source location.